h2/config.py

```diff
@@ -87,13 +93,18 @@ def __init__(self,
                  header_encoding='utf-8',
                  validate_outbound_headers=True,
                  normalize_outbound_headers=True,
-                 validate_inbound_headers=True):
+                 validate_inbound_headers=True,
+                 logger=None):
         self.client_side = client_side
         self.header_encoding = header_encoding
         self.validate_outbound_headers = validate_outbound_headers
         self.normalize_outbound_headers = normalize_outbound_headers
         self.validate_inbound_headers = validate_inbound_headers
 
+        default_logger = logging.getLogger(__name__)
```

Comment:
So I think the problem here is that if somebody else attaches a logger that does I/O to `logging.getLogger(__name__)`, we suddenly have logging appearing in h2 for not entirely obvious reasons.

 > In terms of replacing this with a DummyLogger, would your approach just be to make a object that inherits from the logger, and makes all the logging level calls no-ops?

That sounds sensible to me. If you look at [the source code for the `Logger` class](https://hg.python.org/cpython/file/tip/Lib/logging/__init__.py#l1419), it looks like all the logging goes through the `_log` method. So I think it would be enough to no-op that method, i.e.,

```python
class DummyLogger(Logger):
    def _log(self, *args, **kwargs):
        pass
```

Plus a docstring, tests, etc.